src/components/reactcomponents/AddNewModal.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component, PropTypes } from 'react';
+import SaveFormMutation from '../mutations/SaveFormMutation';
+import  { modernEnvironment } from '../../environment.js'
+
+// A stateless component for rendering the label and input box in the modal dialog box.
+class InputBox extends Component {
+
+  constructor(props: Object) {
+    super(props);
+  }
+
+  render() {
+    const  {
+      type,
+      label,
+      name,
+      placeholder,
+    } = this.props;
+    return (
+      <div>
+        <label htmlFor={name}>{label}</label>
+        <input
+          name={name}
+          placeholder={placeholder}
+          type={type}
+        />
+      </div>
+    );
+  }
+};
+
+// // InputBox.propTypes = {
+//   label: PropTypes.string.isRequired,
+//   name: PropTypes.string.isRequired,
+//   placeholder: PropTypes.string.isRequired,
+//   type: PropTypes.string.isRequired
+// };
+
+/**
+ * Component for Modal Dialog.
+ * Contains the form that allows users to add a new publication to the data.
+ **/
+
+export default class AddNewModal extends Component {
+
+  constructor(props) {
+    super(props);
+    console.log(this.props);
+    this.saveForm = this.saveForm.bind(this);
+  }
+
+  /**
+   * Function to call the close method in enclosing Component when Close icon/background overlay is clicked.
+   * @param e Javascript click event
+   **/
+  closeDialog(e) {
+    e.preventDefault();
+    if (this.props.onClose) {
+      this.props.onClose();
+    }
+  }
+
+  saveForm(e) {
+    e.preventDefault();
+    const form = document.getElementById('newPublication');
```

Comment:
shouldn't be using `document.getElementById` in React. Try this instead (see lines 45 and 20): https://github.com/cabloo/react-graphql-test/pull/1/files#diff-1fe327fbc14708c5fd931b635c387069R45